budgetportal/csv_gen.py

```diff
@@ -0,0 +1,35 @@
+import csv
+from django.http import StreamingHttpResponse
+
+
+class Echo:
+    """An object that implements just the write method of the file-like
+    interface.
+    """
+
+    def write(self, value):
+        """Write the value by returning it, instead of storing in a buffer."""
+        return value
+
+
+def generate_csv_response(data):
+    """
+    :param data: dict
+    :return: StreamingHttpResponse object
+    """
+    response = StreamingHttpResponse(streaming_content=iter_items(data, Echo()),
+                                     content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename=\"openspend-dataset.csv\"'
+
+    return response
+
+
+def iter_items(data, pseudo_buffer):
+    headers_list = data['cells'][0].keys()
```

Comment:
I think make the interface to this module take an argument that is the cell array - so that the caller takes the cell array out of the OpenSpending result - then this module isn't really aware of OpenSpending's response format and that knowledge is a bit more constrained to the json-to-csv view